### DISCUSSION TITLE
User defined filters

### DISCUSSION THREAD

**Discussion by HohMac** [iaincollins/icarus#8] [Ideas & Suggestions]
As I really love this tool if have to spread my ideas to make it even better (at least IMO). ;)

1. Custom defined filters/events
Maybe not the best header, but it describes what I mean.
I´m an Explorer and it would be nice to have the possibility to define filters or custom events on the system list.
For example, you have several stars with 40-50 planets/moons after scanning.
So you  have a hard time going throught all of the data to see what could be interesting for you.
Defining a filter which triggers an event would be really cool (ex.: earth like worlds -> line in system list flashes green or other cool effect)
Same when searching a special material or high G worlds.
The core function of this is also delivered by "ED Observatory" or "Observatory", but I don´t want to use 8 different tools and it doesen´t look good either.

2. Trigger on signal sources 
Maybe it´s the same as point 1., but I´m not sure.
Would love to have a popup or some kind of visual hint when a special signal source is detected while scanning or entering a system.

That´s it for now, keep the good work going; I´m curious to see what´s coming next :)

**Comment by alyyasser** [comment]
i would love this feature too. user defined filters or highlighter would be nice. im using Elite ObservatoryCore with my own custom script right now.

**Comment by iaincollins** [comment]
Hi there!

Thank you!

1. I'm actually planning an explicit view for explorers to highlight high value items to scan or interesting bodies in a system to explore, based on their properties (e.g. earthlike, atmosphere, volcanic activity, botanical life, composition, gravity, rings, proximity to other bodies, etc).

    I'm planning on taking inspiration from tools like ED Observatory and EDDI for what to use as rules (I use EDDI alongside ICARUS Terminal and like it a lot), but I'm very happy if folks are able to expand here on the sorts of things they'd like - or even just post screenshots of things they like in other tools.

2. That's an interesting idea, I haven't really thought about that.

    I wonder if that would work well as filter on the list view (the way the in game navigation menu handles it) or if list of signals should be better in their view in the Navigation panel.

**Comment by alyyasser** [comment]
this is what i have in my ObservatoryCore:
![image](https://user-images.githubusercontent.com/16476111/162664021-0d4d8a47-f53c-4e2a-88f3-4943d5de907b.png)

what i tracked:
- whether the star scoopable or not
- ELW
- WW
- AW
- terraformable high metal content planet/body and whether it were landable or not
- terraformable rocky body and whether it were landable or not
- non-mapped metal rich body and whether it were landable or not
- non-mapped class 2 gas giant

most of it already covered in ICARUS with its icons/indicator, landable or not, terraformable or not, body/planet type
